homeassistant/components/simplisafe/__init__.py

```diff
@@ -517,11 +519,28 @@ def _async_process_new_notifications(self, system: SystemType) -> None:
 
         self._system_notifications[system.system_id] = latest_notifications
 
-    async def _async_websocket_on_connect(self) -> None:
+    async def _async_websocket_connect(self) -> None:
         """Define a callback for connecting to the websocket."""
         if TYPE_CHECKING:
             assert self._api.websocket
-        await self._api.websocket.async_listen()
+
+        should_reconnect = True
+
+        try:
+            await self._api.websocket.async_connect()
+            await self._api.websocket.async_listen()
+        except asyncio.CancelledError:
+            should_reconnect = False
+        except WebsocketError as err:
+            LOGGER.error("Failed to connect to websocket: %s", err)
+        except Exception as err:  # pylint: disable=broad-except
+            LOGGER.error("Unknown exception while connecting to websocket: %s", err)
+
+        if should_reconnect:
+            LOGGER.info("Disconnected from websocket; reconnecting")
+            self._websocket_reconnect_task = self._hass.async_create_task(
+                self._async_websocket_connect()
+            )
```

Comment:
You should empty out the task if it's going to be done.

```suggestion
            )
        else:
            self._websocket_reconnect_task = None
```